File: Desenvolvimento/server/src/ApplicationCore/Entities/Assunto.cs
using ApplicationCore.Interfaces.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace ApplicationCore.Entities
{
    public class Assunto : IBaseEntity<long>
    {
        public long Id { get; set; }
        public string Nome { get; set; }
        public bool Ativo { get; set; }
        public string HashArquivoModelo { get; set; }
        public string Orientacoes { get; set; }

        public long EmpresaId { get; set; }
        public virtual Empresa Empresa { get; set; }

        public long? FluxoId { get; set; }
        public virtual Fluxo Fluxo { get; set; }

        public virtual ICollection<AssuntoArquivo> AssuntoArquivos { get; set; }
    }
}

File: Desenvolvimento/client/src/@fuse/types/models/assunto.ts
import { Empresa } from './empresa';
import { Setor } from './setor';
import { BaseModel } from '../base-model';
import { AssuntoArquivo } from './assunto-arquivo';
import { Fluxo } from './fluxo';

export class Assunto extends BaseModel {
    constructor(
        public id: number,
        public nome?: string,
        public ativo?: boolean,
        public hashArquivoModelo?: string,
        public orientacoes?: string,

        public empresaId?: number,
        public empresa?: Empresa,

        public setorProcessoFisicoId?: number,
        public setorProcessoFisico?: Setor,

        public setorProcessoVirtualId?: number,
        public setorProcessoVirtual?: Setor,

        public fluxoId?: number,
        public fluxo?: Fluxo,

        public assuntoArquivos?: AssuntoArquivo[],
    ) { super(); }
}
